template/9-design.md

```diff
@@ -1,36 +1,68 @@
-# Design and Implementation
+## Design and Implementation
 
 ## Frontend
 
-*List the key libraries, languages, components used by the MVP.*
+#### Implementation framework
 
-*If applicable, describe essential screens.*
+The application will be developed for Android using Kotlin. This approach ensures a modern, null-safe and efficient implementation of the app's UI and business logic.
+
+#### Dynamic UI Rendering Strategy
+
+Composed of two primary flows (Admin flow and Crew/Pilot flow), our JetBrains Compose based UI will render different screens depending on the type of user. This routing enables efficient navigation through app features.
+
+- The Admin flow includes a Home screen featuring upcoming flights, User management screen, Chat screen, and a Statistics screen dedicated to finished flights.
+- The Crew/Pilot flow includes a Home screen displaying user-specific upcoming flights, Flight screen handling in-flight procedures, Chat screen, and a Calendar for updating user availability and assigned flights.
+
+#### Framework
+
+Mainly, Jetpack Compose framework is employed for UI realization. Tailored specifically for our immediate needs, the UI is not subjected to constant changes.
 
 ## Backend
 
-*Decompose the MVP into functional blocks.*
+The application relies on Firebase for backend services, creating a frontend-focused environment. The Firebase SDK serves as a communication medium with Firebase.
 
-## Data Model
+### Authentification
 
-*What data are you collecting / managing?*
+The application employs Firebase authentication mechanism. All users are authenticated through Google, Facebook, Microsoft account or an email address.
 
-*How is it organised?*
+### Database
 
-*Where is it stored?*
+Google Cloud Firestore, a No-SQL database, is used for data storage. It supports live updates and limited queries. Firestore is employed both for long-term data storage like user, vehicle, and balloon data and real-time updates like current user position or a flight status. Firestore-managed cache caters to the application's caching needs.
 
-*How is it shared/copied/cached?*
+### Push notification
 
-## Security Considerations
+Firebase Cloud Messaging (FCM) is used for push notifications. Notifications are triggered on receiving a message or the initiation of a flight.
```

Comment:
Really?

Reply:
It is the design of a product, not our current application, so ideally yes, it should have a push notification service.